core/google/cloud/client.py

```diff
@@ -135,7 +135,7 @@ def __init__(self, credentials=None, _http=None):
                     credentials, google.auth.credentials.Credentials)):
             raise ValueError(_GOOGLE_AUTH_CREDENTIALS_HELP)
         if credentials is None and _http is None:
-            credentials = get_credentials()
+            credentials, _ = google.auth.default()
```

Comment:
> Does this match the behavior we're seeing today?

Yes.

----

FWIW, I noticed this as well when making the change but didn't think "fixing" it made sense in this PR.